rustls/src/error.rs

```diff
@@ -1236,7 +1236,35 @@ mod other_error {
     /// Enums holding this type will never compare equal to each other.
     #[allow(clippy::exhaustive_structs)]
     #[derive(Debug, Clone)]
-    pub struct OtherError(pub Arc<dyn StdError + Send + Sync>);
+    pub struct OtherError(Arc<dyn StdError + Send + Sync>);
+
+    impl OtherError {
+        /// Create a new `OtherError` from any error type.
+        pub fn new(err: impl StdError + Send + Sync + 'static) -> Self {
+            Self(Arc::new(err))
+        }
+
+        /// Create a new `OtherError` from any type that implements `fmt::Display`.
```

Comment:
Not sure about this; I think people who actually need it can put this entirely in their program and that will be a hint that their error types should impl Error?